### PR TITLE
Lockless repo delete

### PR DESCRIPTION
**Pull request by bogeblad**
Optimized so read of repositories is not made with Castor to avoid any chance of locks

### CODE REVIEW

File: src/java/org/infoglue/cms/controllers/kernel/impl/simple/RepositoryController.java [hunk 1]
```diff
@@ -699,6 +702,58 @@ public List getRepositoryVOList() throws ConstraintException, SystemException, B
...
+				repoVO.setRepositoryId(new Integer(rs.getString(1)));
+				repoVO.setName(rs.getString(2));
+				repoVO.setDescription(rs.getString(3));
+				repoVO.setDnsName(rs.getString(4));
+				repoVO.setIsDeleted(rs.getBoolean(5));
+				
+				logger.info("Found:" + repoVO);
+				repositoryVOList.add(repoVO);
+			}
+			rs.close();
```
Comment (peterjaric) [inline src/java/org/infoglue/cms/controllers/kernel/impl/simple/RepositoryController.java]: The close calls will not be executed if an exception is thrown. Shouldn't they be placed in a `finally` clause instead?

File: src/java/org/infoglue/cms/controllers/kernel/impl/simple/RepositoryController.java [hunk 2]
```diff
@@ -699,6 +702,58 @@ public List getRepositoryVOList() throws ConstraintException, SystemException, B
...
+			psmt.close();
+		}
+		catch(Exception e)
```
Comment (peterjaric) [inline src/java/org/infoglue/cms/controllers/kernel/impl/simple/RepositoryController.java]: Catching the Exception base class loses information of what might have gone wrong, when reading the code. Wouldn't it be better to handle the real types of exceptions?

File: src/java/org/infoglue/cms/controllers/kernel/impl/simple/RepositoryController.java [hunk 3]
```diff
@@ -699,6 +702,58 @@ public List getRepositoryVOList() throws ConstraintException, SystemException, B
...
+		{
+			logger.error("Problem getting repo list: " + e.getMessage(), e);
+		}
+		
+		CacheController.cacheObject("repositoryCache", key, repositoryVOList);
```
Comment (peterjaric) [inline src/java/org/infoglue/cms/controllers/kernel/impl/simple/RepositoryController.java]: Should the list be cached even if there was an exception and repositoryVOList is null?

**Comment by bogeblad** [comment]
I updated the pull request with both not caching if exception and better cleanup.

**Pull request merged**